### ISSUE TITLE
Need support globs outside of the project cone

### ISSUE THREAD

**Issue by lifengl** [dotnet/sdk#1115]
By default, project items outside of the project cone are not shown in the solution explorer, unless they have link metadata.  To show files added through glob, the developer must add globs like this inside the project:
```xml
   <Compile Include="..\Shared\**\*.cs">
      <Link>ExternalFiles\%(RecursiveDir)\%(Filename)%(Extension)</Link>
    </Compile>
```
But this is hard to remember and hard to use.  The idea is to create something in the .targets file, and apply Link metadata to items with <LinkBase> metadata, so inside the project, it can be:
```xml
   <Compile Include="..\Shared\**\*.cs">
      <LinkBase>ExternalFiles</LinkBase>
    </Compile>
```
or

```xml
   <Compile Include="..\Shared\**\*.cs">
      <LinkBase></LinkBase>
    </Compile>
```
If it maps to the project root folder directly.

**Comment by davkean** [comment]
@livarcocc We need this for 15.3 - to go along with the feature where we've enabled customers to view files outside of their project cone.

**Comment by livarcocc** [comment]
@davkean Is the ask here that the SDK should add this metadata automatically to Compile Include items?

cc @dsplaisted @nguerrera

**Comment by davkean** [comment]
Yes, @dsplaisted is plugged in. I suspect it's more than Compile, it will be for all items that we know about.

**Comment by nguerrera** [comment]
@davkean Is LinkBase going to be required for the feature you're adding or is it just going to make it nicer to use than the long form of putting Link based on metadata?

**Comment by dasMulli** [comment]
If this is implemented, could `LinkBase` be defaulted to `\`?
It's currently a bit weird to be able to include items (with or without globs) from outside the project cone only to find out that they won't follow the same publishing / copy-to-output rules. e.g. Content items won't be assigned target paths by MSBuild and thus not being picked up by publish (see [this SO question](http://stackoverflow.com/questions/43569821/dotnet-core-publish-include-exclude-dir-in-output/43611163#43611163)).

**Comment by davkean** [comment]
@nguerrera The latter.

**Comment by nguerrera** [comment]
Is there a definitive list of the items that should honor LinkBase?

**Comment by srivatsn** [comment]
The project system will only show Compile, AdditionalFiles, None, Content, EmbeddedResource in the tree. (See [here ](https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/CSharp.ProjectItemsSchema.xaml#L12-L13) and [here](https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ProjectItemsSchema.xaml#L61-L63)). It sounds like we should add to these by default.

**Comment by srivatsn** [comment]
Fixed by #1246
